### PR TITLE
F add orchestration

### PR DESCRIPTION
**Pull request by mbfrahry**
Adds an orchestration block to the instance resource

### CODE REVIEW

File: opc/resource_instance.go [hunk 1]
```diff
@@ -260,6 +260,50 @@ func resourceInstance() *schema.Resource {
...
+						"desired_state": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"active",
+								"inactive",
+								"suspend",
+							}, true),
+						},
+						"object_label": {
```
Comment (scross01) [inline opc/resource_instance.go]: Suggest changing this to just `label` for consistency with other label fields

**Comment by mbfrahry** [comment]
```
OPC_LOG=1 make testacc TEST=./opc TESTARGS="-run=TestAccOPCOrchestratedInstance"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./opc -v -run=TestAccOPCOrchestratedInstance -timeout 120m
=== RUN   TestAccOPCOrchestratedInstance_Basic
--- PASS: TestAccOPCOrchestratedInstance_Basic (132.74s)
=== RUN   TestAccOPCOrchestratedInstance_BasicTwoInstance
--- PASS: TestAccOPCOrchestratedInstance_BasicTwoInstance (140.34s)
=== RUN   TestAccOPCOrchestratedInstance_sharedNetworking
--- PASS: TestAccOPCOrchestratedInstance_sharedNetworking (146.47s)
=== RUN   TestAccOPCOrchestratedInstance_ipNetwork
--- PASS: TestAccOPCOrchestratedInstance_ipNetwork (298.50s)
=== RUN   TestAccOPCOrchestratedInstance_ipNetworkIsDefaultGateway
--- PASS: TestAccOPCOrchestratedInstance_ipNetworkIsDefaultGateway (306.01s)
=== RUN   TestAccOPCOrchestratedInstance_storage
--- PASS: TestAccOPCOrchestratedInstance_storage (334.38s)
=== RUN   TestAccOPCOrchestratedInstance_noBoot
--- PASS: TestAccOPCOrchestratedInstance_noBoot (4.20s)
=== RUN   TestAccOPCOrchestratedInstance_inactive
--- PASS: TestAccOPCOrchestratedInstance_inactive (23.67s)
=== RUN   TestAccOPCOrchestratedInstance_activeToInactive
--- PASS: TestAccOPCOrchestratedInstance_activeToInactive (138.56s)
=== RUN   TestAccOPCOrchestratedInstance_inactiveToActive
--- PASS: TestAccOPCOrchestratedInstance_inactiveToActive (148.44s)
=== RUN   TestAccOPCOrchestratedInstance_activeToSuspend
--- PASS: TestAccOPCOrchestratedInstance_activeToSuspend (119.75s)
PASS
ok  	github.com/terraform-providers/terraform-provider-opc/opc	1793.095s
```

File: opc/resource_orchestrated_instance.go [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+			"version": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceOPCOrchestratedInstanceCreate(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[DEBUG] Resource state: %#v", d.State())
```
Comment (vancluever) [inline opc/resource_orchestrated_instance.go]: Is this line printing anything? State would be pretty much empty at this point, save configuration.

File: opc/resource_orchestrated_instance_test.go [hunk 3]
```diff
@@ -0,0 +1,528 @@
...
+func testAccGetInstanceAttr(instance, attr string) string {
+	return fmt.Sprintf("%s.%s", instance, attr)
+}
+
+func testAccOrchestrationBasic(rInt int) string {
+	return fmt.Sprintf(`
+  resource "opc_compute_orchestrated_instance" "test" {
+    name        = "test_orchestration-%d"
+    desired_state = "active"
+		instance {
```
Comment (vancluever) [inline opc/resource_orchestrated_instance_test.go]: Indentation here could use a fix - run these (and other examples) through `terraform fmt`?

File: website/docs/r/opc_compute_orchestrated_instance.html.markdown [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+layout: "opc"
+page_title: "Oracle: opc_compute_orchestrated_instance"
+sidebar_current: "docs-opc-resource-orchestrated-instance"
+description: |-
+  Creates and manages an Orchestration containing a number of instances in an OPC identity domain.
+---
+
+# opc\_compute\_orchestrated\_instance
+
+The ``opc_compute_orchestrated_instance`` resource creates and manages an orchestration containing a number of
```
Comment (vancluever) [inline website/docs/r/opc_compute_orchestrated_instance.html.markdown]: Double backtick here

File: opc/resource_orchestrated_instance.go (hunk 2 above)
Comment (mbfrahry) [reply opc/resource_orchestrated_instance.go]: I believe there is benefit here in cases where the orchestrated instance is dependent on other resources.

File: opc/resource_orchestrated_instance_test.go (hunk 3 above)
Comment (mbfrahry) [reply opc/resource_orchestrated_instance_test.go]: Good call. Got those for this PR. I'll submit another PR later for fixing the other ones

**Comment by mbfrahry** [comment]
```
OPC_LOG=1 make testacc TEST=./opc TESTARGS="-run=estAccOPCOrchestratedInstance"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./opc -v -run=estAccOPCOrchestratedInstance -timeout 120m
=== RUN   TestAccOPCOrchestratedInstance_Basic
--- PASS: TestAccOPCOrchestratedInstance_Basic (131.14s)
=== RUN   TestAccOPCOrchestratedInstance_Persistent
--- PASS: TestAccOPCOrchestratedInstance_Persistent (135.27s)
=== RUN   TestAccOPCOrchestratedInstance_BasicTwoInstance
--- PASS: TestAccOPCOrchestratedInstance_BasicTwoInstance (139.33s)
=== RUN   TestAccOPCOrchestratedInstance_sharedNetworking
--- PASS: TestAccOPCOrchestratedInstance_sharedNetworking (178.35s)
=== RUN   TestAccOPCOrchestratedInstance_ipNetwork
--- PASS: TestAccOPCOrchestratedInstance_ipNetwork (314.82s)
=== RUN   TestAccOPCOrchestratedInstance_ipNetworkIsDefaultGateway
--- PASS: TestAccOPCOrchestratedInstance_ipNetworkIsDefaultGateway (140.72s)
=== RUN   TestAccOPCOrchestratedInstance_storage
--- PASS: TestAccOPCOrchestratedInstance_storage (161.95s)
=== RUN   TestAccOPCOrchestratedInstance_noBoot
--- PASS: TestAccOPCOrchestratedInstance_noBoot (5.00s)
=== RUN   TestAccOPCOrchestratedInstance_inactive
--- PASS: TestAccOPCOrchestratedInstance_inactive (26.13s)
=== RUN   TestAccOPCOrchestratedInstance_activeToInactive
--- PASS: TestAccOPCOrchestratedInstance_activeToInactive (138.08s)
=== RUN   TestAccOPCOrchestratedInstance_inactiveToActive
--- PASS: TestAccOPCOrchestratedInstance_inactiveToActive (135.01s)
=== RUN   TestAccOPCOrchestratedInstance_activeToSuspend
--- PASS: TestAccOPCOrchestratedInstance_activeToSuspend (145.59s)
PASS
ok  	github.com/terraform-providers/terraform-provider-opc/opc	1651.423s```

**Pull request merged**